generator/internal/descriptor_utils.cc

```diff
@@ -210,13 +211,79 @@ std::string DefaultIdempotencyFromHttpOperation(
       std::exit(1);
   }
 }
+
+std::string FormatClassCommentsFromServiceComments(
+    google::protobuf::ServiceDescriptor const& service) {
+  google::protobuf::SourceLocation service_source_location;
+  if (service.GetSourceLocation(&service_source_location) &&
+      !service_source_location.leading_comments.empty()) {
+    return absl::StrReplaceAll(
+        absl::StrCat(
+            "/**\n *",
+            absl::StrReplaceAll(service_source_location.leading_comments,
+                                {{"\n", "\n * "}}),
+            "\n", " */"),
+        {{"*  ", "* "}});
+  }
+  GCP_LOG(FATAL) << __FILE__ << ":" << __LINE__ << ": " << service.full_name()
```

Comment:
Why does this need to be fatal? Would returning empty string be reasonable instead?